covid19_sfbayarea/data/hospitals.py

```diff
@@ -20,37 +21,119 @@
 BAYPD_META = "This data was pulled from the data.ca.gov CKAN Data API"
 
-logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+# =======================
+# Data Manipulation Utils
+# =======================
+
+
+def truncate_ts(ts: str) -> str:
+    """Truncate a timestamp to an ISO 8601-formatted date"""
+    trunc_ts = parse(ts).date().isoformat()
+    return trunc_ts
+
 
+def convert_null(record: Dict) -> Dict:
+    """Convert any null values to -1"""
+    for k, v in record.items():
+
+        if v is None:
+            record[k] = -1
+
+        else:
+            continue
 
-def get_county(county: str) -> Dict:
-    """Return data just for the selected county. Include field notes.
-    This is basically an alias for `get_timeseries()` provided for consistency.
+    return record
+
+
+def floats_to_ints(record: Dict) -> Dict:
+    """Convert zero-point floats for numeric fields to ints"""
+    fields: List = [
+        "all_hospital_beds",
+        "hospitalized_covid_confirmed_patients",
+        "hospitalized_covid_patients",
+        "hospitalized_suspected_covid_patients",
+        "icu_available_beds",
+        "icu_covid_confirmed_patients",
+        "icu_suspected_covid_patients",
+    ]
+
+    for field in fields:
+        val = record.get(field)
+
+        if val is None:
+            continue
+
+        else:
+            record[field] = int(val)
+
+    return record
+
+
+# ===================
+# Data Transformation
+# ===================
+
+
+def standardize_data(record: Dict) -> Dict:
+    """Transform certain data fields to make them conform to BAPD style
+
+    Specifically:
+    - truncate timestamps to ISO 8601-formatted dates
+    - remove "rank" field, if it exists
+    - convert all 'null' values to -1
+    - cast zero-point floats as int
+
+    Also, the key 'todays_date' is converted to 'report_date' for clarity.
     """
-    data = get_timeseries(county)
+    record["report_date"] = truncate_ts(record.pop("todays_date"))
+    record.pop("rank", None)
+    record = convert_null(record)
+    record = floats_to_ints(record)
 
-    return data
+    return record
 
 
-def get_timeseries(county: str = "all") -> Dict:
-    """Fetch all pages of timeseries data from API endpoint"""
-    ts_data: Dict[str, Union[str, List]] = {}
-    timeseries: List[Dict[str, Any]] = []
-    series_standardized: List[Dict[str, Any]] = []
+def process_data(series: List, counties: List) -> Dict:
+    """Transform the timeseries data (a list of dicts) into a dict
+    with keys for each county name, and a list of dicts with records
+    for that particular county"""
+    processed_series: Dict[str, Union[str, Any]] = {}
+
+    for county in counties:
+        county_name = friendly_county(county)
+        county_records = [
+            standardize_data(record) for record in series
+            if record.get("county") == county_name
+        ]
+
+        processed_series[county_name] = county_records
```

Comment:
If the goal of re-organizing this as a `dict` was consistency with other output we generate, the keys should be the code-friendly county names, not the human friendly ones (the idea being that, in languages where you can do direct property access, someone can do `hospitals.san_francisco` instead of having to do the less-convenient `hospitals['San Francisco']`).

```suggestion
        county_name = friendly_county(county)
        county_records = [
            standardize_data(record) for record in series
            if record.get("county") == county_name
        ]

        processed_series[county] = county_records
```

Reply:
Thanks for flagging that. I've changed this.